### PR TITLE
[SYCL-MLIR] Drop SYCLFuncRegistry module

### PR DESCRIPTION
**Pull request by victor-eds**
The purpose of this module was never clear. Best guess is that it was an idea for a feature that was never implemented.

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]
It was adding in https://github.com/intel/llvm/commit/3b3ffd90d404f4441765c10d6bf2b578119a93bf. We didn't have `MangledFunctionName` attribute to `sycl.call` and `sycl.constructor`, so we want to have a registry to find the mangled function name, and then find out is not doable, as the functions are templated functions, and there can be infinite amount of them. 

As @etiotto is the author, please wait for his approval.

**Comment by etiotto** [comment]
> It was adding in [3b3ffd9](https://github.com/intel/llvm/commit/3b3ffd90d404f4441765c10d6bf2b578119a93bf). We didn't have `MangledFunctionName` attribute to `sycl.call` and `sycl.constructor`, so we want to have a registry to find the mangled function name, and then find out is not doable, as the functions are templated functions, and there can be infinite amount of them.
> 
> As @etiotto is the author, please wait for his approval.

Yes this is old stuff that can be removed.

**Review by etiotto — APPROVED** [review]

**Pull request merged**
